control-plane/cmd/post-install/kafka_channel_migrator.go

```diff
@@ -0,0 +1,291 @@
+	}
+
+	logger.Infof("New data plane is ready, progressing with the migration")
+
+	err = m.migrateChannelServices(ctx, logger)
+	if err != nil {
+		return err
+	}
+
+	err = m.migrateConfigmap(ctx, err, logger)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (m *kafkaChannelMigrator) migrateChannelServices(ctx context.Context, logger *zap.SugaredLogger) error {
+	newDispatcherExternalName := network.GetServiceHostname(NewChannelDispatcherServiceName, system.Namespace())
+
+	logger.Infof("Starting migration of channel services to new dispatcher service: %s.", newDispatcherExternalName)
+
+	// Sample YAML for a consolidated KafkaChannel service:
+	// apiVersion: v1
+	//kind: Service
+	//metadata:
+	//  labels:
+	//    messaging.knative.dev/role: kafka-channel
+	// ...
+	//  name: kafka-channel-kn-channel
+	//  namespace: default
+	//  ownerReferences:
+	//  - apiVersion: messaging.knative.dev/v1beta1
+	//    ...
+	//spec:
+	//  externalName: kafka-ch-dispatcher.knative-eventing.svc.cluster.local
+	//  sessionAffinity: None
+	//  type: ExternalName
+	//status:
+	//  loadBalancer: {}
+
+	// KafkaChannel service (the one that is created per channel instance) in consolidated channel
+	// only has the label "messaging.knative.dev/role: kafka-channel".
+	// Get those services and redirect them to new dispatcher.
+	kafkaChannelServiceLabels := map[string]string{
+		// See https://github.com/knative-sandbox/eventing-kafka/blob/23602e84d707b8c106c93ed9b12737263488ad18/pkg/channel/consolidated/reconciler/controller/resources/service.go#L70
+		"messaging.knative.dev/role": "kafka-channel",
+	}
+
+	// cont takes care of results pagination.
+	// there might be more resources than a single rest call can return, so we need to think about the pagination.
+	cont := ""
+	first := true
+	lsOpts := metav1.ListOptions{LabelSelector: labels.SelectorFromSet(kafkaChannelServiceLabels).String(), Continue: cont}
+	for first || cont != "" {
+		kafkaChannelServiceList, err := m.k8s.CoreV1().
+			Services(corev1.NamespaceAll).
+			List(ctx, lsOpts)
+		if err != nil {
+			return fmt.Errorf("failed to list KafkaChannel services: %w", err)
+		}
+		for _, svc := range kafkaChannelServiceList.Items {
+
+			patch := []byte(fmt.Sprintf(`[{"op":"replace", "path": "/spec/externalName", "value": "%s"}]`, newDispatcherExternalName))
+
+			logger.Infof("Patching service %s/%s with the patch: %s.", svc.Namespace, svc.Name, patch)
+
+			_, err := m.k8s.CoreV1().
+				Services(svc.Namespace).
+				Patch(ctx, svc.Name, types.JSONPatchType, patch, metav1.PatchOptions{})
+			if err != nil {
+				// just log and continue with other services
+				logger.Errorf("error while patching KafkaChannel service %s/%s: %w", svc.Namespace, svc.Name, err)
+			}
+		}
+		cont = kafkaChannelServiceList.Continue
+		first = false
+	}
+	return nil
+}
+
+func (m *kafkaChannelMigrator) migrateConfigmap(ctx context.Context, err error, logger *zap.SugaredLogger) error {
```

Comment:
I think that `err` as a parameter isn't needed, maybe it is a leftover from some automatic refactoring.

Reply:
> maybe it is a leftover from some automatic refactoring.
LOL yes it is